comunicatii/src/main/java/code4romania/czl/scrapers/comunicatii/ScraperComunicatii.java

```diff
@@ -26,216 +26,282 @@
-        List<ComunicatiiBean> outputList = new ArrayList<ComunicatiiBean>();
-        for (Element link : links) {
-            ComunicatiiBean cBean = new ComunicatiiBean();
-            outputList.add(cBean);
-            ComunicatiiBean.Document cDoc = new ComunicatiiBean.Document();
-            extractDocument(link, cBean, cDoc);
-
-            link.textNodes().forEach(linknode -> System.out.println("link node: " + linknode.toString()));
-
-            System.out.println(cBean);
-        }
-    }
-
-    private boolean isSeparator(Element e){
-        e.text();
-        return false;
-    }
-    private void processRelatedElements(List<Element> elements){
-     }
-
-    private static void extractDocument(Element link, ComunicatiiBean cBean, ComunicatiiBean.Document cDoc) {
-        String docUrl = link.attr("href");
-        cDoc.setUrl(docUrl);
-        try {
-            cDoc.setType(docUrl.substring(docUrl.lastIndexOf('.')));
-
-        }catch (Exception e){
-            e.printStackTrace();
-        }
-        cBean.getDocuments().add(cDoc);
-    }
-
-    public static String calculateFeedbackDays(String dataPublicare, String dataFeedback){
-        // Converting date to Java8 Local date
-        DateTimeFormatter fmt = DateTimeFormatter.ofPattern("dd.MM.yyyy");
-        try {
-            LocalDate startDate = LocalDate.parse(dataPublicare, fmt);
-            LocalDate endDate = LocalDate.parse(dataFeedback, fmt);
-            // Range = End date - Start date
-            Long range = ChronoUnit.DAYS.between(startDate, endDate);
-
-            return range.toString();
-        }catch (Exception e){
-            e.printStackTrace();
-        }
-        return "0";
-    }
-
-    public static String deductType(String description){
-        String type ="";
-        if (description.contains("Proiect de Lege")){
-            type = type + "LEGE";
-        }
-        if (description.contains("Ordonanță") || description.contains("Ordonanţă")){
-            type = type + "OG";
-        }
-        if (description.contains("Hotărâre")){
-            type = type + "HG";
-        }
-        return  type;
-    }
-
-    public static final MediaType JSON
-            = MediaType.parse("application/json; charset=utf-8");
-
-    static OkHttpClient client = new OkHttpClient();
-
-    static String post(String url, String json) throws IOException {
-        RequestBody body = RequestBody.create(JSON, json);
-        Request request = new Request.Builder()
-                .addHeader("Authorization", "token comunicatii-very-secret-key")
-                .addHeader("Content-Type", "application/json; charset=utf-8")
-                .url(url)
-                .post(body)
-                .build();
-        Response response = client.newCall(request).execute();
-        return response.body().string();
-    }
+	}
+
+	private static void extractDate(ComunicatiiBean cBean) {
+		Pattern p = Pattern.compile(DATE_REGEX);
```

Comment:
In majoritatea cazurilor `Pattern`-urile pot fi compilate in prealabil si refolosite.